src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -114,84 +101,86 @@ public bool ContentSet
         /// <summary>
         /// Generates the <see cref="TagHelperOutput"/>'s start tag.
         /// </summary>
-        /// <returns><c>string.Empty</c> if <see cref="TagName"/> is <c>null</c> or whitespace. Otherwise, the
+        /// <returns><c>Null</c> if <see cref="TagName"/> is <c>null</c> or whitespace. Otherwise, the
         /// <see cref="string"/> representation of the <see cref="TagHelperOutput"/>'s start tag.</returns>
         public string GenerateStartTag()
         {
             // Only render a start tag if the tag name is not whitespace
             if (_isTagNameNullOrWhitespace)
             {
-                return string.Empty;
+                return null;
             }
 
-            var sb = new StringBuilder();
-
-            sb.Append('<')
-              .Append(TagName);
+            var startTag = new StringBuilder();
+            startTag.Append("<");
```

Comment:
in general, don't change _style_ of existing code to something you prefer (i.e. just because).  we'll end up churning the code back and forth for no reason.  here, most of us prefer to chain when we can.